src/gax/src/error/core_error.rs

```diff
@@ -291,6 +289,7 @@ impl Error {
     pub fn authentication(source: CredentialsError) -> Self {
         Self {
             kind: ErrorKind::Authentication(source),
+            source: None,
```

Comment:
This looks strange. Seems like `ErrorKind::Authentication` should be a unit variant and we put the credential error into `source`.

Reply:
Eventually we may want to expose the `is_transient()` attribute of `CredentialsError`, but we can change things back when that happens.

Reply:
Ah, I see. We could do like:

```
enum AuthErrorKind {
  Transient,
  Permanent,
}
```

```
Authentication(AuthRetryLoopStatus)
```

to simplify our internals a tad.